drivers/dma/dma_stm32.c

```diff
@@ -196,20 +211,26 @@ static int dma_stm32_get_periph_increment(enum dma_addr_adj increment,
 static int dma_stm32_configure(struct device *dev, u32_t id,
 			       struct dma_config *config)
 {
+
 	struct dma_stm32_data *data = dev->driver_data;
-	struct dma_stm32_stream *stream = &data->streams[id];
 	const struct dma_stm32_config *dev_config =
 					dev->config->config_info;
 	DMA_TypeDef *dma = (DMA_TypeDef *)dev_config->base;
+
+	id = id - STREAM_OFFSET;
```

Comment:
hum, quite ugly to have an expression in the middle of the variable declarations. Well, as you want, it seems that lately declaring variable not at the beginning of a code block is now accepted (it's awful ...).

Reply:
changed outside the variable declaration